core/rfb.js

```diff
@@ -1675,33 +1685,45 @@ RFB.messages = {
         sock.flush();
     },
 
-    pixelFormat: function (sock, bpp, depth, true_color) {
+    pixelFormat: function (sock, depth, true_color) {
```

Comment:
what's the reasoning behind dropping the `bpp` argument here?  I'd prefer to be more explicit here, like we were doing before.

Reply:
We weren't explicit before either as only some of the necessary values were provided and the rest were hard coded (possibly creating conflicting info). The idea here was to make it a bit more consistent in that we derive most fields from what is provided.

Reply:
nice pun (a *bit* more consistent), but I don't think I'm following.  We did hard-code the color locations, but we don't have to do that.  Passing in bpp makes it more explicit that we're choosing this value for pixel size, and this other value for depth.

Reply:
But they're not really independent. There's pretty much only a single correct bpp for each depth value. The function should either:

 a) Be really stupid and just send what is given
 b) Be somewhat high level and ask only for the values it cannot derive itself

Right now we've had some mix of the two. This patch pushes it further along the b) route.

Reply:
ack.